tests/cycling-robot.cpp

```diff
@@ -0,0 +1,218 @@
+#include "aligator/core/traj-opt-problem.hpp"
+#include "aligator/core/traj-opt-data.hpp"
+#include "aligator/core/cost-abstract.hpp"
+#include "aligator/core/stage-data.hpp"
+#include "aligator/utils/mpc-util.hpp"
+#include "aligator/solvers/proxddp/solver-proxddp.hpp"
+#include "aligator/modelling/dynamics/multibody-constraint-fwd.hpp"
+#include "aligator/modelling/dynamics/integrator-semi-euler.hpp"
+#include "aligator/modelling/dynamics/integrator-rk2.hpp"
+#include "aligator/modelling/multibody/frame-velocity.hpp"
+#include "aligator/modelling/multibody/frame-placement.hpp"
+#include "aligator/modelling/costs/sum-of-costs.hpp"
+#include "aligator/modelling/costs/quad-state-cost.hpp"
+#include "aligator/modelling/costs/constant-cost.hpp"
+
+#include <pinocchio/multibody/model.hpp>
+#include <pinocchio/algorithm/joint-configuration.hpp>
+#include <pinocchio/algorithm/frames.hpp>
+#include <pinocchio/algorithm/compute-all-terms.hpp>
+#include <boost/test/unit_test.hpp>
+
+using namespace aligator;
+using context::SolverProxDDP;
+
+namespace
+{
+  pinocchio::Model SimpleModel()
+  {
+    pinocchio::Model model;
+
+    auto const joint_id = model.addJoint(0, pinocchio::JointModelPZ(), pinocchio::SE3::Identity(), "joint");
+    auto const frame_id = model.addJointFrame(joint_id, -1);
+    model.addBodyFrame("link", joint_id, pinocchio::SE3::Identity(), static_cast<int>(frame_id));
+
+    auto const CoM = Eigen::Vector3d::Zero();
+    auto const mass = 0.001;
+    auto const I = (0.001 * Eigen::Matrix3d::Identity()).eval();
+    model.appendBodyToJoint(joint_id, pinocchio::Inertia(mass, CoM, I), pinocchio::SE3::Identity());
+
+    return model;
+  }
+
+  using stage_space_t = proxsuite::nlp::MultibodyPhaseSpace<double>;
```

Comment:
```suggestion
  using StageSpace = proxsuite::nlp::MultibodyPhaseSpace<double>;
```
We usually name typedefs capitalized